lib/iiif_manifest/v3/manifest_builder/placeholdercanvas_builder.rb

```diff
@@ -0,0 +1,56 @@
+module IIIFManifest
+  module V3
+    class ManifestBuilder
+      class PlaceholderCanvasBuilder
+        attr_reader :placeholder_content, :canvas_path, :iiif_placeholder_canvas_factory, :iiif_annotation_page_factory,
+                    :content_builder
+        def initialize(placeholder_content,
+                       canvas_path,
+                       iiif_placeholder_canvas_factory:,
+                       iiif_annotation_page_factory:,
+                       content_builder:)
+          @placeholder_content = placeholder_content
+          @canvas_path = canvas_path
+          @iiif_placeholder_canvas_factory = iiif_placeholder_canvas_factory
+          @iiif_annotation_page_factory = iiif_annotation_page_factory
+          @content_builder = content_builder
+        end
+
+        def build
+          return nil if placeholder_content.nil?
+
+          build_placeholder_canvas
+          attach_content
+
+          placeholder_canvas
+        end
+
+        private
+
+        def path
+          "#{canvas_path}/placeholder"
+        end
+
+        def placeholder_canvas
+          @placeholder_canvas ||= iiif_placeholder_canvas_factory.new
+        end
+
+        def build_placeholder_canvas
+          placeholder_canvas['id'] = path
+          placeholder_canvas['width'] = placeholder_content.width if placeholder_content.width.present?
+          placeholder_canvas['height'] = placeholder_content.height if placeholder_content.height.present?
```

Comment:
Do we need to add `type`, `format`, and `duration` properties to the `placeholder_canvas` or is that all contained within the annotation body?

Reply:
The spec does not say `duration` and `format` are must have properties for the `placeholderCanvas`, and with this current implementation if duration and format are specified then they will be contained within the annotation body as you mentioned.

However, according to the Presentation 3.0 spec `duration` is an optional attribute for a `Canvas`. So we can add `duration` as a property of `placeholder_canvas` if we want. What do you think @cjcolvar ?

But `format` is not allowed as a `Canvas` property as it says [here](https://iiif.io/api/presentation/3.0/#appendices) under 'Technical Properties'.

Reply:
:+1: I think I was getting confused with how this code works and where those values get inserted into the manifest.

Since `duration` is added to regular video canvases I think it should be set here if a value is set on the `DisplayContent`.

Reply:
I'll add that change in :+1: